app/models/tag.rb

```diff
@@ -1023,7 +1033,7 @@ def remove_sub_filters(sub_tag)
   before_update :check_canonical
   def check_canonical
     if self.canonical_changed? && !self.canonical?
-      self.async(:remove_canonical_associations)
+      remove_canonical_associations
```

Comment:
I am pretty sure it would work as it is, what I might suggest is that we deploy this by itself so if it causes issues then we can roll it back without causing any other issues.